runbotics/runbotics-scheduler/src/scheduler-database/process-credential/process-credential.service.ts

```diff
@@ -77,6 +81,46 @@ export class ProcessCredentialService {
         );
     }
 
+    async update(processId: number, user: User, processCredentialsToUpdate: EditProcessCredentialsDto) {
+        await this.processService.checkAccessAndGetById(processId, user);
+
+        const currentProcessCredentialsForTemplate = await this.findAllByProcessId(
+            processId,
+            user,
+            processCredentialsToUpdate.templateId
+        );
+
+        const intersection = _.intersectionBy(
+            processCredentialsToUpdate.credentials,
+            currentProcessCredentialsForTemplate,
+            'id'
+        );
+
+        if (currentProcessCredentialsForTemplate.length !== intersection.length) {
+            throw new BadRequestException();
+        }
+
+        return this.processCredentialRepository.manager.transaction(
+            async (manager) => {
+                return Promise.all(processCredentialsToUpdate.credentials.map(async (credential) => {
+                    const partial: Partial<ProcessCredential> = {};
+
+                    partial.order = credential.order;
+
+                    return manager
+                        .getRepository(ProcessCredential)
+                        .update({ id: credential.id }, partial);
+                }));
+            }
+        )
```

Comment:
kind of strange structure, why use manager?

Reply:
Because it uses transaction - it is needed to update every credential as the same transaction, same contract to avoid situation where something went wrong and one credential was updated and the other not